brew_view/controllers/system_list_api.py

```diff
@@ -42,20 +63,48 @@ def get(self):
         tags:
           - Systems
         """
-        self.logger.debug("Getting all Systems")
-
+        query_set = System.objects.order_by(self.request.headers.get('order_by', 'name'))
+        serialize_params = {'to_string': True, 'many': True}
+
+        include_fields = self.request.headers.get('include_fields')
+        exclude_fields = self.request.headers.get('exclude_fields')
+        include_commands = self.get_query_argument('include_commands', None)
+
+        if include_fields and exclude_fields:
+            raise ModelError("Headers 'include_fields' and 'exclude_fields' "
+                             "cannot exist on the same request")
+        elif include_commands and (include_fields or exclude_fields):
+            raise ModelError("Headers 'include_fields' and 'exclude_fields' "
+                             "cannot be used with 'include_commands'")
+        elif include_fields:
+            include_fields = set(include_fields.split(',')) & self.REQUEST_FIELDS
+            query_set = query_set.only(*include_fields)
+            serialize_params['only'] = include_fields
+        elif exclude_fields:
+            exclude_fields = set(exclude_fields.split(',')) & self.REQUEST_FIELDS
+            query_set = query_set.exclude(*exclude_fields)
+            serialize_params['exclude'] = exclude_fields
+
+        # Deal with include_commands
+        if include_commands:
+            if include_commands.lower() == 'false':
+                query_set = query_set.exclude('commands')
+                serialize_params['include_commands'] = False
+            elif include_commands.lower() == 'count':
+                query_set = query_set.no_dereference()
+
+        # Need to use self.request.query_arguments to get ALL the query args
+        # Once we know the name use get_query_argument to get the decoded value
+        # TODO Handle multiple query arguments with the same key
```

Comment:
I think this is TODONE. We handle it.

Reply:
What I mean here is if you did something like `/api/v1/systems?name=echo&name=complex` it would ignore the `name=echo` part. I'm not sure exactly sure what the best thing to do in that case is - maybe an OR?

I punted on this because I think we have an 80% solution here. We can go further on exposing more of the mongo api but I think if we want to do that we should track that under a separate issue.